#!/usr/bin/env python
PACKAGE = "hop_detection"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


gen.add("light_max_aspect_ratio", double_t, 0, "light_max_aspect_ratio", 0.1,   0.0,    1.0)
gen.add("light_min_aspect_ratio", double_t, 0, "light_min_aspect_ratio", 0.67,  0.0,    1.0)
gen.add("light_min_area",         int_t,    0, "light_min_area",         100,   0,   307200)
gen.add("light_max_area",         int_t,    0, "light_max_area",         61440, 0,   307200) # 640*480
gen.add("light_max_angle",        double_t, 0, "light_max_angle",        45.0,  0.0,  180.0)
gen.add("light_max_angle_diff",   double_t, 0, "light_max_angle_diff",   30.0,  0.0,  180.0)
gen.add("light_length_ratio",     double_t, 0, "light_length_ratio",     0.5,   0.0,    1.0)

gen.add("armor_max_angle",          double_t, 0, "armor_max_angle",        30.0,  0,      180)
gen.add("armor_min_area",           int_t,    0, "armor_min_area",         250,   0,   307200)
gen.add("armor_max_aspect_ratio",   double_t, 0, "armor_max_aspect_ratio", 0.4,   0,        1)
gen.add("armor_max_pixel_val",      double_t, 0, "armor_max_pixel_val",    150.0, 0,      255)
gen.add("armor_max_stddev",         double_t, 0, "armor_max_stddev",       40.0,  0,      100)

gen.add("armor_blank_max_angle",    double_t, 0, "armor_blank_max_angle",    45.0, 0,     180)
gen.add("armor_blank_min_area",     int_t,    0, "armor_blank_min_area",     250,  0,  307200)
gen.add("blank_light_length_ratio", double_t, 0, "blank_light_length_ratio", 0.3,  0,       1)

exit(gen.generate(PACKAGE, "hop_detection", "HopDetection"))
